### DISCUSSION TITLE
Arguments problem when using convtranspose2d

### DISCUSSION THREAD

**Discussion by AwayFromKeyb0ard** [ultralytics/ultralytics#10204] [Q&A]
Hello,

in the yaml file when i use this **[-1, 1, nn.ConvTranspose2d, [512,512,2,2,0]]** it says it goes out of memory. When i use [-1, 1, nn.ConvTranspose2d, [512,2,2,0]] it works properly but in logs shows in terminal the model configuration its displayed as [-1, 1, nn.ConvTranspose2d, [512,512,2,2,0]]. Is this a bug?

Based on pytorch documentation, the first argument is input channels and the second the output channels.
https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html

Thanks,
Nol

**Comment by pderrenger** [comment]
@N0lanakis hello Nol,

It sounds like you've encountered an issue with specifying parameters for `nn.ConvTranspose2d` in your YAML configuration. The memory problem when using `[512, 512, 2, 2, 0]` likely arises because specifying both input and output channels as `512` significantly increases the number of parameters and thus the memory requirements.

As for the logs displaying `[-1, 1, nn.ConvTranspose2d, [512,512,2,2,0]]` even when you specify `[512, 2, 2, 0]`, it might be related to how the model configuration is parsed or logged. 🤔

Your observation based on the PyTorch documentation is correct regarding the ordering of parameters. It might be helpful to double-check your YAML file to ensure there isn't a typo or formatting issue causing the configuration to revert or display incorrectly. Additionally, ensure you're using the latest version of the software, as updates may have addressed this issue.

If the problem persists, consider opening an issue with a minimal reproducible example on the repository, providing details about your environment (version numbers etc.). This will help the team to inspect the issue closely.

Meanwhile, using `[512, 2, 2, 0]` as a workaround seems valid for your scenario, provided it aligns with your network's architectural requirements.
